python/test/query-tests/run_integration.py

```diff
@@ -39,26 +41,28 @@ class QuerySet(TypedDict):
 
 def _run_with_resources(
         test_data: TestData,
-        fn,
+        fn: Callable[[ContainerProxy], None],
+        unique_name: str
 ) -> None:
     client = CosmosClient(url=TestConfig.host, credential=TestConfig.masterKey)
-    unique_name = test_data[CONTAINER_PROPERTIES][ID]
     db = client.create_database_if_not_exists(id=unique_name)
     try:
-        pk_paths: list[str] = test_data[CONTAINER_PROPERTIES][PARTITION_KEY][PATHS]
-        pk = PartitionKey(path=pk_paths[0])  # single-path only
-        container = db.create_container_if_not_exists(
-            id=unique_name,
-            partition_key=pk,
-            offer_throughput=40000
-        )
-
-        # insert documents
-        for item in test_data[DATA]:
-            container.create_item(body=item)
-
-        # hand control to the caller
-        fn(container)
+        for container_def in test_data[CONTAINERS]:
+        
+            pk_paths: list[str] = container_def[PARTITION_KEY][PATHS]
+            pk = PartitionKey(path=pk_paths[0])  # single-path only
+            container = db.create_container_if_not_exists(
+                id=container_def[ID],
+                partition_key=pk,
+                offer_throughput=40000
+            )
+
+            # insert documents
+            for item in test_data[DATA]:
+                container.create_item(body=item)
+
+            # hand control to the caller
+            fn(container)
```

Comment:
I don't think this is _quite_ right. The idea of the changes is to populate **all** the containers with test data (which you're doing) but then each query specifies which container it runs against. That allows us to run tests against different indexing policies and other container properties without having a lot of similar test files.